lib/useQuery.js

```diff
@@ -222,28 +230,60 @@ function finder(service, params, { queryId, allPages }) {
 
     fetchNext()
 
+    function doFind(skip) {
+      return service.find({
+        ...params,
+        query: {
+          ...(params.query || {}),
+          $skip: skip,
+        },
+      })
+    }
+
+    function resolveOrFetchNext(res) {
+      if (res.data.length === 0 || result.data.length >= result.total) {
+        resolve(result)
+      } else {
+        skip = result.data.length
+        fetchNext()
+      }
+    }
+
     function fetchNext() {
-      service
-        .find({
-          ...params,
-          query: {
-            ...(params.query || {}),
-            $skip: skip,
-          },
-        })
-        .then(res => {
-          result.limit = res.limit
-          result.total = res.total
-          result.data = result.data.concat(res.data)
-
-          if (res.data.length === 0 || result.data.length >= result.total) {
-            resolve(result)
-          } else {
-            skip = res.skip + res.limit
```

Comment:
I think this was wrong initially 😱 but I don't think it would have caused issues.. maybe, it's simpler just to always use the size of the data array